### PR TITLE
[#4432] migration fix

### LINKED ISSUE #4432: Constrain ResourceFile to one per file
**Description of the bug**
ResourceFile currently allows more than one entry to be made for a single file.  This causes bugs in the code when dealing with the file as the code usually expects only one entry.

Steps to reproduce the bug:
I don't know what action causes multiple ResourceFile entries to be created for a single file.  It likely has to do with an interrupted process.  The management command to identify resources with this issue will meet these criteria.

**Expected behavior**
There should only be one ResourceFile per file.  This constraint should be on the db model.

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #4432 

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

File: hs_core/management/utils.py [hunk 1]
```diff
@@ -215,6 +218,48 @@ def check_relations(resource):
                       .format(resource.short_id, r.type, target))
 
 
+def fix_resourcefile_duplicates(dry_run=False, logger=None):
```
Comment (pkdash) [inline hs_core/management/utils.py:221]: This function really did not change how we are accessing the models.

File: hs_core/migrations/0072_remove_duplicate_res_files.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
-from django.core.management import call_command
+import logging
+from django.db import migrations
+from hs_core.management.utils import fix_resourcefile_duplicates
+
 
 def migrate_resourcefiles(apps, schema_editor):
-    call_command('fix_resourcefile_duplicates')
+    logger = logging.getLogger(__name__)
+    fix_resourcefile_duplicates(dry_run=False, logger=logger)
```
Comment (pkdash) [inline hs_core/migrations/0072_remove_duplicate_res_files.py:10]: In the management command we should be accessing the current models (should not be using apps.get_model. I might have made this mistake.

File: hs_core/management/utils.py (hunk 1 above)
Comment (devincowan) [reply hs_core/management/utils.py]: believe we addressed this

File: hs_core/management/utils.py [hunk 3]
```diff
@@ -215,6 +218,52 @@ def check_relations(resource):
...
+    if not logger:
+        logger = logging.getLogger(__name__)
+    # first we remove files with content_type other than the content type for CompositeResource
+    if get_model:
+        CompositeResourceModel = apps.get_model('hs_composite_resource', 'CompositeResource')
+        ResourceFileModel = apps.get_model('hs_core', 'ResourceFile')
+    else:
+        from hs_core.models import ResourceFile as ResourceFileModel
+        from hs_composite_resource.models import CompositeResource as CompositeResourceModel
+    desired_content_type = ContentType.objects.get_for_model(CompositeResourceModel)
```
Comment (pkdash) [inline hs_core/management/utils.py:236]: Use apps.get_model() for ContentTYpe:
`ContentType = apps.get_model('contenttypes', 'ContentType')`

File: hs_core/management/utils.py [hunk 4]
```diff
@@ -15,9 +15,12 @@
 import os
 import time
 
+from django.apps import apps
 from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
```
Comment (pkdash) [inline hs_core/management/utils.py:20]: Bases on the error I was getting on my fix for migration:
Remove this import

Comment (devincowan) [reply hs_core/management/utils.py]: did that in https://github.com/hydroshare/hydroshare/pull/5191/commits/d61680e7597996d4243fdb209a8e11a42ae60cdc thanks

File: hs_core/management/utils.py (hunk 3 above)
Comment (devincowan) [reply hs_core/management/utils.py]: did that in https://github.com/hydroshare/hydroshare/pull/5191/commits/d61680e7597996d4243fdb209a8e11a42ae60cdc thanks

**Review by pkdash — APPROVED** [review]
Hope the manual reordering of migrations works during deployment.

**Pull request merged**